### PR TITLE
Udated @gnosis.pm/safe-service-client to the latest and optimized it usage

### PR DESCRIPTION
**Pull request by shoom3301**
# Do not merge before https://github.com/cowprotocol/cowswap/pull/820

# Summary
1. Updated `@gnosis.pm/safe-service-client` package to the latest version
2. Refactored `useWalletInfo()` hook
3. Created redux store for `gnosisSafe` to avoid multiply HTTP requests to the Gnosis API (`useWalletInfo()` made it)

  # To Test

1. Make a swap via Gnosis safe

  # Background
`@gnosis.pm/safe-service-client` was updated accidentally, when I tried to find the `isReadOnly` flag #311

### CODE REVIEW

File: src/custom/state/orders/mocks.ts
Comment (shoom3301) [inline src/custom/state/orders/mocks.ts:84]: This code is unused

**Review by anxolin — APPROVED** [review]
Great PR! 

I'm happy to see how quickly you got used to the structure of the project! The Updater, and  the simplification of the logic looks great to me.

The PR is still manageable, although is in the fence of being a big PR. 

Maybe in the description, you could also give some context why you needed to add the provider info, and why the library update was required.

We even try to do dependency updates in isolation, with the minimal changes, so its easier to review, but also to make it easier to pin point to the PR introducing an issue if we detect something in the future

File: src/custom/api/gnosisSafe/index.ts
Comment (shoom3301) [reply src/custom/api/gnosisSafe/index.ts:43]: image
Because `Web3Provider` expects a specific type.
And we bind the `send` method to avoid context leak

File: src/custom/state/gnosisSafe/actions.ts
Comment (alfetopito) [inline src/custom/state/gnosisSafe/actions.ts:4]: May I suggest to use [jotai](https://jotai.org/) instead?

I know, mixing state management is a terrible idea.
But it is already mixed :)

It was introduced awhile ago by Uniswap in the Widgets subapp and we have been experimenting to use that in the rest of the app.
It might come a point where we switch over - that is, assuming the experiment is successful and we are happy with the results.

What do you say @anxolin, good place for using it here, no?

**Review by alfetopito — COMMENTED** [review]
Minor comments on the code below.

Regarding the behaviour, it might or might not have started on this PR.

When I load the CowSwap as a Safe app and while I'm not connected, it shows `Read only`.
When I connect a wallet that is a Safe owner, CowSwap remains `read only`.

Are we able to detect that from within the Safe?

**Comment by nenadV91** [comment]
Very nice changes, looks much cleaner. Just to keep in mind that new uni-merge introduces some changes to these parts, for example there is no `library` property anymore, for the reference 
https://github.com/cowprotocol/cowswap/blob/92eb189ef4dffb7fc558509d650b0858d8c0796c/src/custom/hooks/useWalletInfo.ts
https://github.com/cowprotocol/cowswap/blob/92eb189ef4dffb7fc558509d650b0858d8c0796c/src/custom/api/gnosisSafe/index.ts
so I don't know if we want to merge this now or after the uni-merge.

**Comment by shoom3301** [comment]
@nenadV91 thank you for the review!
The refactoring is not urgent, then let's merge it after the uni-merge.

File: src/custom/state/gnosisSafe/actions.ts
Comment (shoom3301) [reply src/custom/state/gnosisSafe/actions.ts:4]: Awesome! It's exactly what I was looking for!
I will add it

**Review by elena-zh — CHANGES_REQUESTED** [review]
Hey @shoom3301 , the app is crashed when navigate to the Account/About/FAQ/CowRunner/MEV Slicer/Terms and Conditions page inside the Safe
image

All the rest functionality related to the Swap form LGTM

File: src/custom/hooks/useWalletInfo.ts
Comment (alfetopito) [inline src/custom/hooks/useWalletInfo.ts:137]: There's a shortcut for that https://jotai.org/docs/utils/use-atom-value

File: src/custom/state/gnosisSafe/updater.tsx
Comment (alfetopito) [inline src/custom/state/gnosisSafe/updater.tsx:13]: There's also a shortcut for that :) 

https://jotai.org/docs/utils/use-update-atom

**Review by alfetopito — APPROVED** [review]
Approved with refactoring suggestions

**Comment by shoom3301** [comment]
https://github.com/cowprotocol/cowswap/pull/870#pullrequestreview-1060400859
@elena-zh looks like it was a temporary problem related to deployment. When you opened the app and after I pushed a new commit it rewrites chunks and you get this error. It shouldn't happen in production.

**Review by elena-zh — APPROVED** [review]
LGTM now!

**Comment by shoom3301** [comment]
https://github.com/cowprotocol/cowswap/pull/929

**Pull request closed without merging**
